src/web/app/pages-session/session-submission-page/session-submission-page.component.ts

```diff
@@ -478,6 +461,59 @@ export class SessionSubmissionPageComponent implements OnInit, AfterViewInit {
         });
   }
 
+  /**
+   * Redirects to login page if feedback session is accessible.
+   *
+   * @param auth The authentication information used in the redirection URL.
+   */
+  redirectToLoginPageIfFeedbackSessionAccessible(auth: AuthInfo): void {
+    this.isFeedbackSessionLoading = true;
+    this.feedbackSessionsService.getFeedbackSession({
+      courseId: this.courseId,
+      feedbackSessionName: this.feedbackSessionName,
+      intent: this.intent,
+      key: this.regKey,
+      moderatedPerson: this.moderatedPerson,
+      previewAs: this.previewAsPerson,
+    }).pipe(finalize(() => {
+      this.isFeedbackSessionLoading = false;
+    }))
+      .subscribe(() => {}, (resp: ErrorMessageOutput) => {
+        this.handleGetFeedbackSessionError(resp, true, auth);
+      });
+  }
+
+  private handleGetFeedbackSessionError(resp: ErrorMessageOutput, isRegisteredUser: boolean, auth?: AuthInfo): void {
```

Comment:
I don't think you need the second parameter. `auth` being present or not is already the same condition.